src/Npgsql/NpgsqlTypes/PostgisTypes.cs

```diff
@@ -105,22 +275,38 @@ internal int GetLen(bool includeSRID)
     }
 
     /// <summary>
-    /// Represents an Postgis 2D Point
+    /// PostGIS base Geometry type with specified coordinate system.
     /// </summary>
-    public class PostgisPoint : PostgisGeometry, IEquatable<PostgisPoint>
+    /// <typeparam name="T">Coordinate type for Geometry</typeparam>
+    public abstract class PostgisGeometry<T> : PostgisGeometry where T : Coordinate2D
     {
-        Coordinate2D _coord;
+    }
+
+    /// <summary>
+    /// PostGIS base Point type with specified coordinate system.
+    /// </summary>
+    /// <typeparam name="T">Coordinate type for Geometry</typeparam>
+    public abstract class PostgisPoint<T> : PostgisGeometry<T> where T : Coordinate2D
+    {
+        internal abstract Coordinate2D Coordinate { get; }
```

Comment:
As I wrote above, I'm not sure I understand this. Why not simply have a distinct `Coordinate` property on each `PostgisPoint` type, exposing the precise coordinate type?

This would also make the `PostgisPoint<T>` abstract base class redundant.

Reply:
Agree with this. Changes made in 2f8519cbd649906d339029636f3fbf7bf30336b0